### PR TITLE
[Inductor] Fix view lowering dropping the view on raw IR nodes

### LINKED ISSUE #189401: [inductor] Unrolled sum reduction produces wrong results for non-contiguous (transposed) inputs after `unsqueeze+expand`
### 🐛 Describe the bug

When `make_pointwise(ops.mul)` + `make_reduction("sum")` operates on non-contiguous (transposed) inputs after `unsqueeze` + broadcast `expand`, the generated Triton kernel uses **wrong stride multipliers** — it assumes contiguous strides for the logical shape instead of preserving the actual transposed strides.

The `ops.dot` + `reduction("dot")` path (native_matmul) is NOT affected because it bypasses the standard tiling heuristic and forces `{y: M, x: N, r0_: K}` tiling unconditionally.

### Root cause

Traced through 4 independent investigations to:

1. **Reduction unrolling** (`ir.py:1770-1787`): When `reduction_numel < unroll_reductions_threshold` (default 8) AND `reduction_type != "dot"`, the reduction is unrolled into a `Pointwise` node via `_unroll_reduction_fn`. The `"dot"` reduction is explicitly exempt.

2. **View.create path difference**: Without `clone`, the input is a `ReinterpretView` with `is_storage_and_layout=True`, so `View.create` uses the `_compute_stride` path. With `clone`, the input becomes a `Pointwise` with `is_storage_and_layout=False`, routing through the more robust `dynamic_reshape_indexer` (symbolic reindexing).

3. **Generated code evidence**: For `mm(64×3, (4×3).T)`, the buggy kernel loads `in_ptr1 + (4*k + x0)` (contiguous strides `(4,1)` for shape `(3,4)`) when the actual transposed strides are `(1,3)`. The correct load pattern is `in_ptr1 + (k + 3*x0)`.

Related to #186850 (closed as non-reproducible, different reproducer).

### Reproducer

```python
import torch

a = torch.randn(64, 3, device='cuda')
b = torch.randn(4, 3, device='cuda')

@torch.compile
def fn(x, y):
    return torch.mm(x, y.T)

ref = torch.mm(a, b.T)
out = fn(a, b)
print(f"max diff: {(ref - out).abs().max().item()}")
# Expected: ~1e-7 (float32 tolerance)
# Actual: ~8.0 (completely wrong)
```

**Note**: This only reproduces when mm is decomposed to `unsqueeze + mul + sum` at the lowering level (e.g., via the small-mm pointwise optimization in PR #189149). Standard cuBLAS mm is not affected.

To reproduce without PR #189149, insert this in `tuned_mm` after `mm_args`:
```python
mat1 = lowerings[aten.unsqueeze](mat1, -1)
mat2 = lowerings[aten.unsqueeze](mat2, 0)
args, _ = transform_args(args=[mat1, mat2], kwargs={}, broadcast=True,
                          type_promotion_kind=None, convert_input_to_bool=False)
return make_reduction("sum")(make_pointwise(ops.mul)(*args), axis=1)
```

### Workaround

`clone` before `unsqueeze` forces the input through the `dynamic_reshape_indexer` path:
```python
mat1 = L.clone(mat1)
mat2 = L.clone(mat2)
```
The clone has zero runtime cost because the scheduler fuses the clone's Pointwise node into the downstream mul kernel.

### Versions

PyTorch 2.14.0a0+git30e161d (viable/strict), CUDA 12.8, H200

cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @muchulee8 @amjames @aakhundov @coconutruben @jataylo @jansel @eellison @yanboliang @ngimel

### PR DESCRIPTION
**Pull request by guan404ming**
## Related Issue

Fixes #189401

## Why

- Compiled `torch.mm` on a transposed input silently returned wrong numbers
- A clone workaround forced extra copies instead of fixing the cause

## How

- `.data` on a `ReinterpretView` returns the base storage and drops the transposed layout, so `View.create` re-derived indices against contiguous strides
- Keep the view when the `view` lowering receives a raw IR node, and widen its annotation to `IRNode`
- Drop the `L.clone` workaround in `tuned_mm`, which only masked the bug by turning the operand into a `Pointwise`
- The lowering-level fix is chosen over wrapping at the call site because `view(ConstantBuffer)` previously raised `AttributeError`

## Test

```
python test/inductor/test_mmdecomp.py
python test/inductor/test_torchinductor.py
```

On an RTX 5090: `test_mmdecomp` is 210 passed with the fix and 54 failed without it. `test_torchinductor` is 3036 tests with one pre-existing failure (`test_tmp_not_defined_issue3_cpu`, fails on main too).

The transposed test now covers lhs/rhs/both and asserts `decompose_mm_pointwise` fired, so it can no longer pass vacuously.

This PR was authored with the assistance of Claude.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by guan404ming** [comment]
Added lhs/both coverage, counter assertions, IRNode annotation; split the skip change.

**Comment by guan404ming** [comment]
The B200 failure is the skip predicate I split out; now #191181.

**Review by eellison — APPROVED** [review]

**Review by mlazos — APPROVED** [review]
